### PR TITLE
Improve docs related to the basis vectors and components

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Better descriptions for `Naxis_vec`, `Ncomponents_vec` and `basis_vector_array`. The existing descriptions are unhelpful.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Documentation change checklist:
- [x] Any updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] If this is a significant change to the readme or other docs, I have checked that they are rendered properly on ReadTheDocs. (you may need help to get this branch to build on RTD, just ask!)

### CODE REVIEW

**Comment by bhazelton** [comment]
@slosar I'd love to get your read of this when you have time.

File: pyuvdata/uvbeam/uvbeam.py [hunk 1]
```diff
@@ -212,11 +218,13 @@ def __init__(self):
...
-            "Beam basis vector components -- directions for which the "
-            "electric field values are recorded in the pixel coordinate system. "
+            "Beam basis vector components, essentially the mapping between the "
+            "directions that the electrical field values are recorded in to the "
+            "directions aligned with the pixel coordinate system (or azimuth/zenith "
+            "angle for HEALPix beams)."
             'Not required if beam_type is "power". The shape depends on the '
             'pixel_coordinate_system, if it is "healpix", the shape is: '
-            "(Naxes_vec, Ncomponents_vec, Npixels), otherwise it is "
+            "(Naxes_vec, Ncomponents_vec=2, Npixels), otherwise it is "
```
Comment (slosar) [inline pyuvdata/uvbeam/uvbeam.py:227]: Why is Ncomponents_vec=2 fixed to 2 if it is not fixed to two above?

File: pyuvdata/uvbeam/uvbeam.py [hunk 2]
```diff
@@ -212,11 +218,13 @@ def __init__(self):
...
             "(Naxes_vec, Ncomponents_vec, Naxes2, Naxes1)"
         )
         self._basis_vector_array = uvp.UVParameter(
```
Comment (slosar) [inline pyuvdata/uvbeam/uvbeam.py:230]: Why is acceptable range (0,1)? Shouldn't it be at least (-1,1)?

**Review by slosar — COMMENTED** [review]
This is in general good, but I think it would be very useful if a few canonical cases were described in the readthedocs coumentation. In particular, what are the settings for these values for
* pixels in (phi,theta), Efield in (phi,theta)
* pixels in (phi,theta), Efield in (x,y,z)
* pixels in healpix, Efield in (phi,theta)?

File: pyuvdata/uvbeam/uvbeam.py (hunk 1 above)
Comment (bhazelton) [reply pyuvdata/uvbeam/uvbeam.py:227]: I was thinking for that the HEALPix we support Ncomponents_vec will always be 2, but I think that is better shown in the examples you suggested, I'll take it out here.

File: pyuvdata/uvbeam/uvbeam.py (hunk 2 above)
Comment (bhazelton) [reply pyuvdata/uvbeam/uvbeam.py:230]: You are correct, it should be (-1,1). Thanks for catching that!

**Comment by bhazelton** [comment]
I added some text and the examples you suggested to the top of the UVBeam tutorial. I'm not 100% sure I have the E-field in (x, y, z) basis vector right, can you see if you agree with my math? And if my new text is clear and helpful?

File: docs/uvbeam_tutorial.rst [hunk 3]
```diff
@@ -8,6 +8,58 @@ models for radio telescopes. UVBeam supports both E-field and power beams
...
+
+  - ``Naxis_vec = 3``
+  - ``Ncomponents_vec = 2``
+  - ``basis_vector_array`` contains the mapping from (x, y, z) to (azimuth, zenith_angle)
+    for the E-field (which is always orthogonal to the radial vector pointing to the
+    pixel). The details of this mapping depend on the choice of how the (x, y, z) axes
+    are labeled. UVBeam uses azimuth running from East to North. If +x points to East,
+    +y points to North and +z points to the zenith:
+
+    - ``basis_vector_array[0, 0, az, za] = sin(az)``
```
Comment (slosar) [inline docs/uvbeam_tutorial.rst:44]:
```suggestion
    - ``basis_vector_array[0, 0, az, za] = -sin(az)``
```
I think this should be negative assuming your phi=0 is E=x and phi=pi/2=N=y (which I think I read somewhere).

File: docs/uvbeam_tutorial.rst [hunk 4]
```diff
@@ -8,6 +8,58 @@ models for radio telescopes. UVBeam supports both E-field and power beams
...
+    - ``basis_vector_array[0, 1, az, za] = cos(az)``
+    - ``basis_vector_array[0, 2, az, za] = 0``
+    - ``basis_vector_array[1, 0, az, za] = cos(za)cos(az)``
+    - ``basis_vector_array[1, 1, az, za] = cos(za)sin(az)``
+    - ``basis_vector_array[1, 2, az, za] = sin(za)``
```
Comment (slosar) [inline docs/uvbeam_tutorial.rst:49]:
```suggestion
    - ``basis_vector_array[1, 2, az, za] = -sin(za)``
```
I also think this should be negative. If basis vectors points towards increasing coordinate, then theta one points down.

**Review by slosar — APPROVED** [review]
I think this is much clearer now, thanks.

File: docs/uvbeam_tutorial.rst (hunk 3 above)
Comment (bhazelton) [reply docs/uvbeam_tutorial.rst:44]: Yes, you're right. thank you!

File: docs/uvbeam_tutorial.rst (hunk 4 above)
Comment (bhazelton) [reply docs/uvbeam_tutorial.rst:49]: Yes, you're right, theta should point down. Thanks!

**Review by jpober — APPROVED** [review]
Looks good to me!

**Pull request merged**